File: app/src/main/java/demo/android/zeno/name/zenokit/MainActivity.java
package demo.android.zeno.name.zenokit;

import android.app.Activity;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.widget.LinearLayout;

import java.util.Arrays;
import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;
import demo.android.zeno.name.zenokit.item.MenuItem;
import kale.adapter.CommonRcvAdapter;
import kale.adapter.item.AdapterItem;
import name.zeno.android.listener.Action1;
import name.zeno.android.presenter.ZActivity;
import name.zeno.android.presenter.ZNav;

public class MainActivity extends ZActivity
{
  @BindView(R.id.rcv_class) RecyclerView rcvClass;

  private List<Class<? extends Activity>>             classes;
  private CommonRcvAdapter<Class<? extends Activity>> adapter;

  @Override
  protected void onCreate(Bundle savedInstanceState)
  {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_main);
    ButterKnife.bind(this);
    init(savedInstanceState);
  }

  private void init(Bundle savedInstanceState)
  {
    Action1<Class<? extends Activity>> onClick = aClass -> ZNav.nav(this, aClass);

    classes = Arrays.asList(TimeButtonActivity.class, ShapeActivity.class);

    adapter = new CommonRcvAdapter<Class<? extends Activity>>(classes)
    {
      @NonNull @Override public AdapterItem createItem(Object type)
      {
        return new MenuItem(onClick);
      }
    };

    rcvClass.setLayoutManager(new LinearLayoutManager(getContext()));
    rcvClass.setAdapter(adapter);
  }

}

File: app/src/main/java/demo/android/zeno/name/zenokit/MainActivity.kt
package demo.android.zeno.name.zenokit

import android.annotation.SuppressLint
import android.app.Activity
import android.os.Bundle
import android.support.v7.widget.LinearLayoutManager
import com.hwangjr.rxbus.annotation.Subscribe
import demo.android.zeno.name.zenokit.item.MenuItem
import io.reactivex.Observable
import kale.adapter.CommonRcvAdapter
import kale.adapter.item.AdapterItem
import kotlinx.android.synthetic.main.activity_main.*
import name.zeno.android.core.nav
import name.zeno.android.data.models.TextData
import name.zeno.android.presenter.ZActivity
import name.zeno.android.third.rxbus.registerRxBus
import name.zeno.android.third.rxbus.rxBus
import name.zeno.android.third.rxjava.RxUtils
import name.zeno.android.util.ZLog
import java.util.concurrent.TimeUnit

class MainActivity : ZActivity() {

  private var classes: List<Class<out Activity>>? = null
  private var adapter: CommonRcvAdapter<Class<out Activity>>? = null

  @SuppressLint("MissingSuperCall")
  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    setContentView(R.layout.activity_main)
    init()
  }

  private fun init() {
    classes = listOf(
        TimeButtonActivity::class,
        ShapeActivity::class,
        CircleIndicatorActivity::class,
        DragLayoutActivity::class
    ).map { it.java }

    adapter = object : CommonRcvAdapter<Class<out Activity>>(classes) {
      val onClick = { clazz: Class<out Activity> -> nav(clazz) }
      override fun createItem(type: Any): AdapterItem<*> = MenuItem(onClick)
    }
    rcv_class.layoutManager = LinearLayoutManager(this)
    rcv_class.adapter = adapter
  }
}
